integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -264,8 +308,142 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
+                s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                  s"Unsupported value: ${handlerType.get} specified for property $TYPE.")
+            }
+        }
+
+        try {
+          val handlerClass: Class[_] = java.lang.Class.forName(handlerClassName)
+          val instance = handlerClass.newInstance().asInstanceOf[
+            CustomIndex[Long, String, java.util.List[Array[Long]]]]
+          instance.init(handler, tableProperties.asJava)
+
+          // TODO Need to check to how to store the instance. This serialization may be incorrect.
+          val bos = new ByteArrayOutputStream()
+          val oos = new ObjectOutputStream(bos)
+          oos.writeObject(instance)
+          oos.flush()
+          tableProperties.put(HANDLER_INSTANCE, Base64.getEncoder().encodeToString(bos.toByteArray))
+        } catch {
+          case ex: ClassNotFoundException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"$handlerClassName class in property $HANDLER_CLASS failed with $ex")
+          case ex@(_: InstantiationError | _: IllegalAccessException) =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Instantiation of class $handlerClassName failed with $ex")
+          case ex: ClassCastException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Failed due to $ex")
+        }
+
+        /* Add target column in sort column */
+        var sortKey = tableProperties.getOrElse(CarbonCommonConstants.SORT_COLUMNS, "")
+        if (!sortKey.isEmpty) {
+          /* Source columns are not allowed to be specified in sort columns. Instead target column
+          is implicitly treated as sort column */
+          sources.foreach { column =>
+            sortKey.split(",").map(_.trim).foreach { key =>
+              if (key.equalsIgnoreCase(column)) {
+                throw new MalformedCarbonCommandException(
+                  s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                    s"Source column: $key is not allowed in ${CarbonCommonConstants.SORT_COLUMNS}" +
+                    s" property. Instead, handler: $handler is implicitly treated as sort column.")
+              }
+            }
+          }
+          sortKey += "," + handler
+        } else {
+          sortKey = handler
+        }
+        tableProperties.put(CarbonCommonConstants.SORT_COLUMNS, sortKey)
+        // TODO Need to convert it to DataType object and pass it to StructField dataType
+        val dataType = tableProperties.get(TYPE)
+        fields += getField(StructField(handler, LongType))
```

Comment:
It has become geohash specific. Can check the handler instance type and collect the data type from handler itself

Reply:
Have refactored it. Currently support only index type of long type.